### ISSUE TITLE
Object literal treatment under ADVANCED_COMPILATION is not clear

### ISSUE THREAD

**Issue by gonzojive** [google/closure-compiler#975]
It's unclear how object literals are treated under ADVANCED_COMPILATION.  The page [tutorial page that explains ADVANCED_COMPILATION](https://developers.google.com/closure/compiler/docs/api-tutorial3) is incomplete.

I believe the following code does not preserve the 'preserveMe' property:
```var x = {preserveMe: false};```

However, this code does:
```var x = {'preserveMe': false};```

Please clarify the compilation semantics in the docs.

**Comment by concavelenz** [comment]
yes, quoted object literal keys must be quoted to be preserved if there isn't not an matching extern declarations.

**Comment by dimvar** [comment]
The example in the section "Inconsistent Property Names" uses an object literal.
